### DISCUSSION TITLE
How to add the PatientID to the PDF file of an print from PrintSCP?

### DISCUSSION THREAD

**Discussion by backsubzero** [fo-dicom/fo-dicom#1246] [Q&A] [answered]
Using the Sample.Desktop "Print SCP" I managed to print to pdf in a clean and perfect way, now I need to save the PDF file name with the PatientID number avoiding confusion in my idea of the process I want to implement, but I can't get this PatientID when I get a print from a workstation or any device on the network, is there any way to get this?

Can anybody help me?

**Comment by gofal** [comment]
Yes, this is a common question that you will find on every DICOM-forum or Stackoverflow.
No, it is not possible.
the printing protocol in DICOM is designed to send rendered pixel data to a printer to print it on some paper or film. All the information is burnt into the pixel data of the image to print, no Patientid or Patientname or studyinstanceuid or seriesinstanceuid in metadata. 
the only chance, which is an ugly and error-prone workaround, would be doing OCR on the image to detect the burnt in text in pixel data.

**Comment by adilt** [comment, marked as answer]
Hi,

gofal is right but there is another way without doing OCR but it depends on modality that send you images!

There is maybe a way to retrieve some informations on one tag called Film Session Label (2000,0050).
Some modalities send PatientsName or/and AccessionNumber or PatientID, some informations (like Fuji with Axon and Eyepix).
So if your Print SCU fill Film Session Label with the PatientID or something else, you win! ;)
